### ISSUE TITLE
`resources` imported but unused

### ISSUE THREAD

**Issue by IamRezaMousavi** [astral-sh/ruff#4250]
`resources.py` must imported to use `Qt resources` in project
resources file will generated by:
`pyrcc5 resources.qrc -o "./resources.py"`

how can I tell ruff to ignore resources import error??

ruff output:
`window.py:22:26: F401 [*] resources imported but unused`

pyproject.toml:
```toml
[tool.ruff]
line-length = 100

[tool.ruff.per-file-ignores]
"__init__.py" = ["F401"]
"resources.py" = ["E501"]
"colorbox.py" = ["E722"]
"test.py" = ["E402", "F401"]
```

ruff version:
`ruff 0.0.264`

**Comment by JonathanPlasse** [comment]
Can you provide a minimal reproducible code example?

**Comment by JonathanPlasse** [comment]
Would adding a noqa be fine for your use case?

**Comment by IamRezaMousavi** [comment]
> Can you provide a minimal reproducible code example?

minimal example code:
https://stackoverflow.com/a/36673994
or see
https://gist.github.com/seanwu1105/4e9362cadedec5353874ef8e28463c2d

**Comment by IamRezaMousavi** [comment]
> Would adding a noqa be fine for your use case?

No I want to see this error for other import code
but ignore all the `resources.py` imports

**Comment by JonathanPlasse** [comment]
I guess a setting could be added to ignore `F401` for a list of imports.

**Comment by IamRezaMousavi** [comment]
> I guess a setting could be added to ignore `F401` for a list of imports.

How?

**Comment by JonathanPlasse** [comment]
It needs to be implemented.

**Comment by JonathanPlasse** [comment]
Do you import `ressources` in a lot of files?
If not the noqa could be added manually.

**Comment by IamRezaMousavi** [comment]
> Do you import `ressources` in a lot of files?
> If not the noqa could be added manually.

Yes

**Comment by IamRezaMousavi** [comment]
Can you explain how can I implement this?

**Comment by JonathanPlasse** [comment]
I can implement it if you want.

**Comment by IamRezaMousavi** [comment]
> I can implement it if you want.

I want to help you and contributing in this project.
If you have no problem!

**Comment by JonathanPlasse** [comment]
To implement this you can look up [CONTRIBUTING.md/#example-adding-a-new-configuration-option](https://github.com/charliermarsh/ruff/blob/main/CONTRIBUTING.md#example-adding-a-new-configuration-option).
`F401` is implemented here.
https://github.com/charliermarsh/ruff/blob/fc63c6f2e260760963c5fdea2d9d4e0cf38ec58a/crates/ruff/src/checkers/ast/mod.rs#L5475-L5626

**Comment by charliermarsh** [comment]
> No I want to see this error for other import code
> but ignore all the resources.py imports

To clarify, you can ignore the unused import _only_ for that line with:

```python
import resources  # noqa: F401
```

That will prevent an unused import error from appearing for that line, but won't prevent other imports from being marked as unused.

**Comment by IamRezaMousavi** [comment]
> That will prevent an unused import error from appearing for that line, but won't prevent other imports from being marked as unused.

Right, but what if I want import like this?
If I just want to ignore _Name:
```python
from ImportFrom import OtherName, _Name
```

I can do like this but it's ugly:
```python
from ImportFrom import OtherName
from ImportFrom import _Name # noqa: F401
```

(This is solved in #4691 )

**Comment by charliermarsh** [comment]
I think the conventional way to format that would be:

```python
from ImportFrom import (
    OtherName,
    _Name,  # noqa: F401
)
```
